### PR TITLE
Stateful mig instance

### PR DESCRIPTION
**Pull request by slevenick**
Support for per-instance configs for stateful managed instance groups.

Fine-grained resource attached to instance group manager. Similar resource for regional instance group manager coming soon

This PR modifies VirtualFields in MM to support types other than booleans. This allows for enums to be set on per-instance configs that control the update strategy for the stateful instances.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_compute_per_instance_config` (Beta)
```

### CODE REVIEW

File: products/compute/api.yaml [hunk 1]
```diff
@@ -7914,7 +7914,127 @@ objects:
...
+        'Official Documentation': 'https://cloud.google.com/compute/docs/instance-groups/stateful-migs#per-instance_configs'
+      api: 'https://cloud.google.com/compute/docs/reference/rest/beta/instanceGroupManagers'
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        kind: 'compute#operation'
+        path: 'name'
+        base_url: 'projects/{{project}}/zones/{{zone}}/operations/{{op_id}}'
+        wait_ms: 1000
+        timeouts: !ruby/object:Api::Timeouts
+          insert_minutes: 6
```
Comment (emilymye) [inline products/compute/api.yaml:7949]: createInstances i assume will need a little more time here? or on deletion

File: products/compute/terraform.yaml [hunk 2]
```diff
@@ -1255,6 +1256,56 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           The Region in which the created address should reside.
           If it is not provided, the provider region is used.
 
+  PerInstanceConfig: !ruby/object:Overrides::Terraform::ResourceOverride
+    id_format: "{{project}}/{{zone}}/{{instance_group_manager}}/{{name}}"
+    mutex: networkEndpoint/{{project}}/{{zone}}/{{instance_group_manager}}
```
Comment (emilymye) [inline products/compute/terraform.yaml:1261]: it's gated against the network endpoint? if its just the IGM, maybe we should rename this mutex

File: products/compute/terraform.yaml [hunk 3]
```diff
@@ -1255,6 +1256,56 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
+        values:
+          - :REPLACE
+          - :RESTART
+          - :REFRESH
+          - :NONE
+        default_value: :NONE
+      - !ruby/object:Api::Type::Enum
+        name: 'most_disruptive_allowed_action'
+        description: |
+          The minimal action to perform on the instance during an update.
```
Comment (emilymye) [inline products/compute/terraform.yaml:1296]: forgot to change this

File: products/compute/terraform.yaml (hunk 2 above)
Comment (slevenick) [reply products/compute/terraform.yaml:1261]: Ah, yeah copied from network endpoint and forgot to change

File: products/compute/api.yaml (hunk 1 above)
Comment (slevenick) [reply products/compute/api.yaml:7949]: I haven't seen it take very long, but happy to bump these up

**Comment by slevenick** [comment]
> We should for sure have at least one test with more than one disk if possible, if the object is a set. Is it supported yet?

Added.

**Pull request merged**